x-pack/plugins/security_solution/server/lib/telemetry/diagnostic_task.ts

```diff
@@ -108,7 +108,9 @@ export class TelemetryDiagTask {
     }
     this.logger.debug(`Received ${hits.length} diagnostic alerts`);
 
-    const diagAlerts: TelemetryEvent[] = hits.map((h) => h._source);
+    const diagAlerts: TelemetryEvent[] = hits.flatMap((h) =>
```

Comment:
Good idea

Reply:
This is a bit of a trick where I'm using `flatMap` to do both filtering and mapping together. It's not the way I would prefer to do it but it fixes our TypeScript issues where TypeScript cannot remove `undefined` cleanly with a `filter`